### PR TITLE
hash: _update never returns anything

### PR DESCRIPTION
**Pull request by dcousens**
This isn't a breaking change - `/hash` is not being used externally AFAIK,  GitHub search shows no usage.

### CODE REVIEW

**Comment by dcousens** [comment]
I also removed the pointless `_s` state

**Comment by dcousens** [comment]
Ready to merge

File: hash.js [hunk 1]
```diff
@@ -15,56 +14,62 @@ Hash.prototype.update = function (data, enc) {
...
-    for (var i = 0; i < ch; i++) {
-      buffer[(s % this._blockSize) + i] = data[i + f]
+    for (var i = 0; i < count; i++) {
+      block[rem + i] = data[offset + i]
     }
+    offset += count
+    accum += count
 
-    s += ch
-    f += ch
```
Comment (dcousens) [inline hash.js]: Single letter variables,  so annoying

**Comment by dominictarr** [comment]
is this just a style change? the title is "update never returns anything" but it returns `this`? am i missing something?

**Comment by dcousens** [comment]
@dominictarr sorry, I meant `_update`

> is this just a style change? 

yes

**Comment by dcousens** [comment]
Tests are failing in Node 5/6 from `npm i buffer`? Hmph

**Comment by dominictarr** [comment]
okay, sure. is old node failing because the default encoding changed from binary to utf8? that is my bet

**Comment by dcousens** [comment]
@dominictarr it is failing at installing `├─┬ buffer@2.8.2 invalid`,  would pass if https://github.com/crypto-browserify/sha.js/pull/44 was merged

**Comment by calvinmetcalf** [comment]
Shouldn't we switch to safe buffer?

**Comment by dcousens** [comment]
@calvinmetcalf point!  I'll do that

**Comment by calvinmetcalf** [comment]
we should probably be using safe-buffer

**Comment by dcousens** [comment]
`safe-buffer` now in use,  good to merge :+1:

**Comment by dcousens** [comment]
Pushed up 1ac913b as part of this

**Pull request merged**